File: 04-2-One2NineAppWithRoom_kt2/app/src/main/java/com/example/one2nineappwithroom_kt2/repository/ScoreRepository.kt
package com.example.one2nineappwithroom_kt2.repository

import kotlinx.coroutines.flow.Flow

class ScoreRepository(private val scoreDao: ScoreDao) {
    suspend fun insertScore(score: Score) = scoreDao.insert(score)

    suspend fun deleteAllScores() = scoreDao.deleteAllScores()

    fun getAllScores(): Flow<List<Score>> = scoreDao.getAllScores()

    suspend fun getBestScore(): Score = scoreDao.getBestScore()
}

File: flutter_projects/04-1-one2nineapp_fl/lib/persistence/scoreRepository.dart

import 'package:one2nineapp_fl/persistence/appDatabase.dart';
import 'package:one2nineapp_fl/persistence/score.dart';
import 'package:one2nineapp_fl/persistence/scoreDao.dart';

class ScoreRepository {

  late AppDatabase database;
  late ScoreDao scoreDao;

  Future<void> initialize() async {
    database = await $FloorAppDatabase.databaseBuilder('app_database.db').build();
    scoreDao = database.scoreDao;
  }

  Future<void> insert(Score score) => scoreDao.insert(score);

  Stream<List<Score>> getAllScores() => scoreDao.getAllScores();

  Future<void> deleteAllScores() => scoreDao.deleteAllScores();

  Future<Score?> getBestScore() => scoreDao.getBestScore();
}